File: telemetry-service/src/main/java/net/blueshell/telemetry/repository/TelemetryRepository.java
package net.blueshell.telemetry.repository;

import jakarta.validation.constraints.NotNull;
import net.blueshell.db.BaseRepository;
import net.blueshell.telemetry.model.Telemetry;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface TelemetryRepository extends BaseRepository<Telemetry, UUID> {

    @Query("SELECT n FROM Telemetry n ORDER BY n.createdAt DESC")
    @Override
    Page<Telemetry> findAll(@NotNull Pageable pageable);

    @Query("SELECT n FROM Telemetry n ORDER BY n.createdAt DESC")
    @Override
    List<Telemetry> findAll();
}

File: api/src/main/kotlin/net/blueshell/api/domain/telemetry/persistence/repository/TelemetryRepository.kt
package net.blueshell.api.domain.telemetry.persistence.repository

import jakarta.validation.constraints.NotNull
import net.blueshell.api.shared.repository.BaseRepository
import net.blueshell.api.telemetry.persistence.Telemetry
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface TelemetryRepository : BaseRepository<Telemetry, Long> {
    @Query("SELECT n FROM Telemetry n ORDER BY n.createdAt DESC")
    override fun findAll(@NotNull pageable: @NotNull Pageable): Page<Telemetry>

    @Query("SELECT n FROM Telemetry n ORDER BY n.createdAt DESC")
    override fun findAll(): MutableList<Telemetry>
}
